From 492afbed124fc716f6d5523eafeb8d06e69741a6 Mon Sep 17 00:00:00 2001
From: lei zhilin <leizhilin@eswincomputing.com>
Date: Mon, 11 Aug 2025 15:29:25 +0800
Subject: [PATCH 434/541] WIN2030-18987:feat(2d):hae use opp table freq

Changelogs:
    1.system need hae use opp table freq rather than cycle cacul frequency, it will cause system ipa controll err;
    2.optimize hae devfeq code;

Change-Id: I8acb59e966523b6d3ea70be78ab095258e1dbc74
Signed-off-by: lei zhilin <leizhilin@eswincomputing.com>
---
 .../eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c     | 5 ++---
 .../eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.h     | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
index 98ce08bd0b46..801df92efc33 100644
--- a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
+++ b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
@@ -3479,7 +3479,7 @@ gc_df_exit(gckGALDEVICE Device)
 }
 #endif
 
-int gckGALDEVICE_GetDevFreqInfo(struct device *dev, struct devfreq_dev_status *stat)
+int gckGALDEVICE_GetHardwareLoad(struct device *dev, struct devfreq_dev_status *stat)
 {
     gctINT32 i = 0;
     gckHARDWARE hardware = NULL;
@@ -3511,12 +3511,11 @@ int gckGALDEVICE_GetDevFreqInfo(struct device *dev, struct devfreq_dev_status *s
             // IPA use, not user hardware->load to prevent clock change too frequncy;
             stat->busy_time = 1024; // hardware->load
             stat->total_time = 1024; // 100
-            stat->current_frequency = hardware->threadMcClk;
             return 0;
         }
     }
 
-    dev_err(dev, "hae get dev freq info failed!\n");
+    dev_err(dev, "hae hardware load no device found!\n");
 
     return -1;
 }
diff --git a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.h b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.h
index 62e1294174ce..3e5b54563185 100644
--- a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.h
+++ b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.h
@@ -161,7 +161,7 @@ gckGALDEVICE_Construct(gcsPLATFORM *Platform, const gcsMODULE_PARAMETERS *Args,
 gceSTATUS
 gckGALDEVICE_Destroy(gckGALDEVICE Device);
 
-int gckGALDEVICE_GetDevFreqInfo(struct device *dev, struct devfreq_dev_status *stat);
+int gckGALDEVICE_GetHardwareLoad(struct device *dev, struct devfreq_dev_status *stat);
 
 static gcmkINLINE gckKERNEL
 _GetValidKernel(gckGALDEVICE Device)
-- 
2.54.0

